### PR TITLE
Add on-device Hugging Face Core ML inference

### PR DESCRIPTION
**Pull request by ales27pm**
## Résumé

- ajoute une véritable inférence Hugging Face/Core ML sur iPhone avec `ales27pm/Dolphin3.0-CoreML` épinglé à `95671cf9a2f56d2a381816ae264cd9aae335d96f`;
- télécharge et vérifie les huit artefacts Dolphin par taille/SHA-256, les exclut récursivement des sauvegardes iCloud, puis compile et met en cache le `.mlpackage` sur l’iPhone;
- exécute le contrat Dolphin stateful exact sous iOS 18 (`inputIds`, `causalMask`, `keyCache`, `valueCache`, `logits`) avec préremplissage segmenté, streaming, échantillonnage, annulation et protections thermiques;
- expose un choix explicite serveur/iPhone sans repli silencieux;
- isole les historiques locaux par compte et protège les courses d’hydratation, de changement de backend et de rafraîchissement serveur.

## Correctifs de revue

Les constats encore reproductibles ont été corrigés, notamment :

- cycle de vie déterministe des promesses React Native et annulation fiable avant/après l’accusé natif;
- annulation native avant `dispose()` asynchrone;
- validation des options mutables et rejet explicite des logits entièrement non finis;
- troncature du prompt au niveau des jetons;
- exclusion iCloud propagée et vérifiée sur le snapshot et tous les fichiers du modèle;
- nettoyage des erreurs lors du déchargement;
- hydratation du backend avant toute requête serveur, garde des callbacks différés et isolation des conversations locales;
- préservation des messages arrivés pendant un rafraîchissement d’historique;
- blocage des changements de backend pendant une génération;
- états UI cohérents pour simulateur, historique local persistant, métadonnées d’erreur et initialisation unique;
- vérification réelle des membres PBX Sources/Frameworks et de toutes les cibles iOS >= 18;
- `Package.resolved` du workspace Xcode avec 27 révisions exactes;
- couverture ESLint étendue aux sources et tests TypeScript/JavaScript.

Le signalement affirmant que `.github/workflows/ci.yml` avait une source de vérité générée n’est pas étayé par le dépôt : aucun générateur, CODEOWNERS ni règle d’ownership correspondante n’existe. Le préflight natif est donc une étape explicite du job mobile maintenu dans ce workflow; le hook `pretest` a été retiré afin que Jest demeure exécutable indépendamment des shells natifs.

## Validation

- `npm run doctor:native`
- `npm run typecheck`
- `npm run lint` — 0 erreur (2 avertissements préexistants)
- `npm test -- --runInBand` — 8 suites, 41 tests
- `git diff --check`
- inventaire Hugging Face vérifié à la révision épinglée : 8 fichiers, tailles et SHA-256 exacts, 1 825 812 981 octets
- test d’intégration du tokenizer Dolphin épinglé réussi avec les vrais fichiers Hub
- dernière exécution SwiftPM disponible avant les garde-fous finaux : 27 tests, 26 réussis et 1 fixture Hub optionnelle ignorée; les changements Swift finaux ont reçu une relecture statique supplémentaire
- 27/27 pins du lock SwiftPM identiques au résolveur
- validation statique du workspace Xcode et du projet PBX

## Validation appareil restante

Le runner Linux ne fournit ni Xcode ni iPhone physique. Avant fusion, effectuer un build Xcode 16+ pour iOS 18 et un essai sur appareil : téléchargement interrompu/repris, premier chargement, génération, annulation immédiate, passage en arrière-plan, pression mémoire et charge thermique soutenue.

## Summary by CodeRabbit

* **New Features**
  * Added on-device Core ML inference: model download/verification, status, streaming generation, cancellation, unload, and deletion.
  * Added inference backend selection (server vs on-device) with a dedicated settings flow and backend-aware chat UI.
  * Composer now supports send gating and in-flight cancellation; on-device requests show Core ML finish details and token metrics.
* **Bug Fixes**
  * Improved chat history synchronization and deduping to avoid overwriting newer realtime/local messages.
  * Preserved local/on-device conversation content during backend/session transitions; migrated legacy local turns to the correct persisted owner.
* **Tests**
  * Expanded Core ML/inference-store/chat-store and UI tests for event ordering, lifecycle, unavailable-device behavior, and cancellation.
* **Chores**
  * Added native wiring validation checks and updated CI; adjusted iOS deployment target and lint/test scripts.

## État de la CI distante

Le run GitHub Actions [29631856774](https://github.com/ales27pm/monGARS/actions/runs/29631856774) du commit `097809b` échoue encore dans `Determine impacted areas` avant toute étape; tous les jobs dépendants, dont le job mobile, sont ignorés. Ce blocage d’infrastructure est identique aux runs précédents et précède l’exécution du code modifié. CodeRabbit termine toutefois avec succès sur ce commit.
## Suivi de revue — commit 3240c74

- le test App affirme maintenant que `Conversation locale conservée` est réellement rendue, tout en conservant l’assertion de l’état modèle indisponible;
- la fusion de l’historique déduplique, par empreinte de tour, les paires serveur apparues pendant la requête et également renvoyées sous d’autres IDs;
- l’annulation est vérifiée entre les tentatives de tokenisation/troncature du prompt;
- la recherche descendante exacte de ce commit est maintenant supersédée par la troncature sensible aux frontières BPE du commit `1ce6f7f`, qui ne retourne qu’une séquence réellement retokenisée dans son template.

## Suivi de revue — commit 97a884f

Les 14 constats du dernier lot étaient encore valides et ont tous été corrigés :

- le préflight cible précisément `MonGARSMobile`, lit les réglages effectifs via Xcode lorsque disponible et résout sinon les xcconfig référencés;
- `doctor:native` est explicite dans la CI, tandis que les tests Jest n’en dépendent plus via `pretest`;
- le sampler distingue un `topK` mutable invalide d’un modèle corrompu, le préfill dérive toutes ses formes d’une constante et les erreurs thermiques/modèle ne sont plus masquées par un drapeau d’annulation;
- la troncature directe de ce commit est maintenant supersédée par un recul logarithmique qui retokenise chaque suffixe candidat avec son template complet;
- les opérations natives s’exécutent dans un worker indépendant du module React Native, avec une cible d’événements faible; le store JS isole aussi chaque cycle de génération périmé;
- chaque chargement revalide réellement les SHA-256 des artefacts, même si taille et date de modification sont inchangées;
- la récupération d’état efface les erreurs périmées, la recherche vide conserve l’état « aucun résultat » et les tours locaux v4 sont migrés vers leur propriétaire persisté;
- la fusion d’historique préserve et réconcilie les tours en vol, y compris avec un tour temps réel intercalé et des empreintes répétées.

## Suivi de revue — commit 1ce6f7f

Sur les huit nouveaux constats, six étaient entièrement valides et deux partiellement valides :

- l’annulation ambiante du test était intentionnelle pour protéger contre une régression vers `Task.isCancelled`; le test affirme maintenant explicitement cette précondition avant de vérifier qu’une erreur concrète demeure prioritaire;
- aucun second fichier de tests dupliqué n’existait, mais le classificateur l’était réellement; `InferenceError.isCancellation` est maintenant l’unique implémentation partagée par le package et le bridge;
- l’invalidation conserve l’opération active et met le déchargement en attente, de sorte que le nettoyage ne peut plus courir en parallèle avec un téléchargement ou chargement annulé;
- la preuve SHA-256 est réutilisée uniquement en mémoire après une vérification cryptographique réussie, puis invalidée avant toute mutation, réparation ou suppression; chaque nouveau processus vérifie encore intégralement le modèle;
- les suffixes de prompt sont décodés puis retokenisés dans le template complet avec un recul logarithmique et des points d’annulation, ce qui respecte les fusions BPE aux frontières;
- le test React attend désormais la valeur différée avec `findByText`;
- le préflight demande explicitement les réglages `iphoneos`, ignore les variantes simulateur non pertinentes et précise que le contrat iOS 18 concerne `MonGARSMobile`, seul target qui lie MonGARSCoreML.

## Suivi Hugging Face/Core ML — commit 70dbb8d

- remplacement de l’ancien artefact Qwen par `Dolphin3.0-Llama3.2-3B-stateful-int4.mlpackage`, source `dphn/Dolphin3.0-Llama3.2-3B@392a6f57223e7ccfe6ef4ebdb2ff101a42d57364`;
- contrat runtime exact : contexte 2048, requête/préremplissage maximal 512, vocabulaire 128 258, deux états KV `[28, 1, 8, 2048, 128]`, arrêts `128256/128001/128008/128009`;
- compilation Core ML asynchrone sur appareil, cache `.mlmodelc` dérivé et exclu d’iCloud, reconstruction unique si le cache est invalide, espace libre minimal annoncé de 5 Go;
- validation dynamique des formes `[1, q, 128258]`, rejet de tout logit non fini, validation défensive des options mutables et rétention explicite du backing `MLMultiArray`;
- tests ajoutés pour le découpage causal, les masques, le manifeste, le sampler et l’intégration du vrai tokenizer.

Aucune exécution du modèle 3B sur iPhone physique n’est attestée par cet environnement ni par le rapport de publication du modèle. Le build Xcode et les essais mémoire/jetsam, thermique, annulation et premier chargement sur un appareil iOS 18 restent donc obligatoires avant fusion.

## Suivi de revue — commit 097809b

Les deux constats étaient encore valides et ont été corrigés minimalement :

- l’alerte de téléchargement affiche maintenant les exigences exactes du manifeste Dolphin : 1 825 812 981 octets (1,83 Go) à télécharger et 5 000 000 000 octets (5,00 Go) libres requis;
- l’échec de création de l’énumérateur récursif d’exclusion iCloud propage désormais une `CocoaError.fileReadUnknown` au lieu de déclarer implicitement le succès;
- un test UI couvre les valeurs de l’alerte; les 8 suites/42 tests Jest, TypeScript, ESLint, `doctor:native` et `git diff --check` passent localement.

### CODE REVIEW

File: mobile-app/ios/MonGARSCoreML/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+- source model `Qwen/Qwen3-1.7B` at revision
+  `70d244cc86ccca08cf5af4e1e306ecf908b1ad5e`
+- 512-token context, with up to 192 new tokens per response
+- 64-token stateful prefill and one-token stateful decode
+- 16 segmented logits outputs for a 151,936-token vocabulary
+
+The downloader requests only the files listed in `ModelManifest.swift`, pins
+the Hub revision, and verifies the size and SHA-256 digest of the tokenizer and
+compiled model payload before loading them. Files live under Application
+Support and are excluded from iCloud backup.
```
[the remark this answers is not in the document]
Comment (ales27pm) [reply mobile-app/ios/MonGARSCoreML/README.md]: Corrigé dans l’arbre courant : l’exclusion iCloud est appliquée au répertoire du snapshot et à chacun des artefacts attendus après téléchargement/réparation. Toute erreur de `setResourceValues` est propagée, donc la préparation ne peut plus annoncer un succès si la politique de sauvegarde échoue.

File: mobile-app/ios/MonGARSCoreML/Package.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  platforms: [
+    .iOS(.v18),
+    .macOS(.v15),
+  ],
+  products: [
+    .library(name: "MonGARSCoreML", targets: ["MonGARSCoreML"]),
+  ],
+  dependencies: [
+    .package(
+      url: "https://github.com/huggingface/swift-transformers.git",
```
[the remark this answers is not in the document]
Comment (ales27pm) [reply mobile-app/ios/MonGARSCoreML/Package.swift:16]: Corrigé : le workspace Xcode contient maintenant `ios/MonGARSMobile.xcworkspace/xcshareddata/swiftpm/Package.resolved`, avec les 27 révisions SwiftPM résolues et vérifiées.

File: mobile-app/src/screens/ChatScreen.tsx [hunk 3]
```diff
@@ -350,6 +432,11 @@ const ChatScreen: React.FC = () => {
             onDraftChange={setDraft}
             quickActions={quickActions}
             sending={loading}
+            canSend={backend === 'on-device' ? localReady : Boolean(session)}
+            embeddingAvailable={
+              backend === 'server' && Boolean(settings.embedServiceUrl)
+            }
+            onCancel={backend === 'on-device' ? cancelGeneration : undefined}
```
[the remark this answers is not in the document]
Comment (ales27pm) [reply mobile-app/src/screens/ChatScreen.tsx:443]: Corrigé : `setInferenceBackend` refuse le changement tant que le chat charge, qu’un `activeRequestId` existe ou que le statut est `generating`; le store d’inférence couvre aussi la phase de démarrage et la promesse en attente. La réponse et son contrôle d’annulation restent donc attachés au backend propriétaire.

File: mobile-app/ios/CoreMLInference/CoreMLInferenceModule.swift [hunk 4]
```diff
@@ -0,0 +1,917 @@
...
+  ) {
+    // Reserved for consent/network policy options. Model selection stays pinned
+    // to MonGARSModelManifest and is never accepted from JavaScript.
+    _ = options
+    let promise = CoreMLBridgePromise(resolve: resolve, reject: reject)
+    let operation = ScheduledOperation(
+      id: UUID().uuidString,
+      kind: .prepare,
+      priority: .utility,
+      run: { [self] in
```
[the remark this answers is not in the document]
Comment (ales27pm) [reply mobile-app/ios/CoreMLInference/CoreMLInferenceModule.swift]: Corrigé dans `97a884f` : les opérations asynchrones capturent un `OperationWorker` indépendant plutôt que le module. La livraison d’événements passe par une cible qui ne référence le module que faiblement; l’invalidation détache/annule l’opération active sans cycle de rétention du pont.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -340,6 +340,12 @@ jobs:
         working-directory: mobile-app
         run: npm run typecheck
 
+      - name: Validate native project wiring
```
[the remark this answers is not in the document]
Comment (ales27pm) [reply .github/workflows/ci.yml]: Constat vérifié mais non applicable au dépôt : aucune source génératrice, règle CODEOWNERS ni règle d’ownership pour `ci.yml` n’existe. Le workflow est maintenu à la main; `doctor:native` y est maintenant une étape mobile explicite, et la justification est consignée dans la description de la PR.

File: mobile-app/ios/MonGARSCoreML/Tests/MonGARSCoreMLTests/InferenceCoordinatorTests.swift [hunk 6]
```diff
@@ -0,0 +1,29 @@
+#if canImport(CoreML)
```
[the remark this answers is not in the document]
Comment (ales27pm) [reply mobile-app/ios/MonGARSCoreML/Tests/MonGARSCoreMLTests/InferenceCoordinatorTests.swift]: Vérifié partiellement : aucun second fichier ni second jeu de tests équivalent n’existe dans le dépôt. En revanche, le classificateur était bien dupliqué entre le coordinator et le bridge. Le commit `1ce6f7f` centralise désormais la logique dans `InferenceError.isCancellation`; les deux consommateurs utilisent cette implémentation unique, couverte par un seul jeu de tests.
